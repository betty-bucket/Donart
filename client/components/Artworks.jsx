import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { getArt } from '../api'

import Artwork from './ArtworkItem'

export default function Artworks () {
  const [artworks, setArtworks] = useState({
    artworks: []
  })
  useEffect(() => {
    getArt()
      .then((res) => {
        setArtworks(res)
      })
      .catch((error) => {
        console.log('error: ', error.message)
      })
  }, [])
  return (
    <>
      <div className='artworkDisplay'>
        <div className='displayTitle'>
          <hr />
          <h2>Browse Artworks</h2>
          <hr />
        </div>
        <div className="card-flex-wrapper column is-three-quarters">
          {artworks.artworks.map((artwork) =>
            <Artwork key={artwork.id} artwork={artwork}/>
          )}
        </div>
      </div>

    </>
  )
}

// Display picture of artwork
// Display name of artwork
// display price of artwork
// Display charity artwork is going towards
