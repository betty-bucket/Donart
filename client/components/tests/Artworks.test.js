import React from 'react'
import { Route } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Artworks from '../Artworks'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Route>
        <Artworks />
      </Route>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
