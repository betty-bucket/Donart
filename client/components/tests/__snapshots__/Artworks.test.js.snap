// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="artworkDisplay"
>
  <div
    className="displayTitle"
  >
    <hr />
    <h2>
      Browse Artworks
    </h2>
    <hr />
  </div>
  <div
    className="card-flex-wrapper column is-three-quarters"
  />
</div>
`;
